File: csharp/202_Happy_Number.cs
public class Solution {
    HashSet<int> checkedSum = new HashSet<int>();
    public bool IsHappy(int n) {
        if (n == 1) return true;

        int sum = 0;
        while (n > 0) {
            sum += (int)Math.Pow(n % 10, 2);
            n /= 10;
        }

        if (checkedSum.Contains(sum)) {
            return false;
        } else {
            checkedSum.Add(sum);
            return IsHappy(sum);
        }
    }
}

File: cpp/202_Happy_Number.cpp
class Solution {
    unordered_set<int> checkedSum;
public:
    bool isHappy(int n) {
        if (n == 1) return true;

        long sum = 0;
        while (n > 0) {
            sum += pow(n % 10, 2);
            n /= 10;
        }
        
        if (checkedSum.count(sum)) {
            return false;
        }
        else {
            checkedSum.insert(sum);
            return isHappy(sum);
        }
    }
};